pkg/reconciler/service/service.go

```diff
@@ -382,5 +383,12 @@ func (c *Reconciler) reconcileRoute(ctx context.Context, service *v1alpha1.Servi
 	existing.Spec = desiredRoute.Spec
 	existing.ObjectMeta.Labels = desiredRoute.ObjectMeta.Labels
 	existing.ObjectMeta.Annotations = desiredRoute.ObjectMeta.Annotations
+
+	rc := route.Status.GetCondition(apis.ConditionReady)
+	if rc != nil && rc.Status == corev1.ConditionFalse {
+		logger.Infof("Updating ObservedGeneration")
+		existing.Status.ObservedGeneration++
```

Comment:
```suggestion
		existing.Status.ObservedGeneration = existing.Generation
```

We might miss spec updates and catch up to multiple changes.

Reply:
I don't think the change you have here is right.  I think this should be taken care of by line 188

Reply:
@mattmoor I think line 188 is updating the `ObservedGeneration` for `service.Status`, versus here we are updating the `ObservedGeneration` for `route.Status`

Reply:
Oh, I misread.

Status fields for resources should only be modified by the reconciler _for that resource_.  So this is likely the change we want, but we want it in the **Route** reconciler, not the Service reconciler.